src/main/java/org/spongepowered/common/world/gen/SpongeChunkGenerator.java

```diff
@@ -449,23 +449,27 @@ public BlockPos getNearestStructurePos(World worldIn, String structureName, Bloc
         } else if ("EndCity".equals(structureName)) {
             target = MapGenEndCity.class;
         }
-        if (target == null) {
-            return null;
-        }
-        for (GenerationPopulator gen : this.genpop) {
-            if (target.isInstance(gen)) {
-                return ((MapGenStructure) gen).getNearestStructurePos(worldIn, position, p_180513_4_);
+        if (target != null) {
+            for (GenerationPopulator gen : this.genpop) {
+                if (target.isInstance(gen)) {
+                    return ((MapGenStructure) gen).getNearestStructurePos(worldIn, position, p_180513_4_);
+                }
             }
         }
+
         if (this.baseGenerator instanceof SpongeGenerationPopulator) {
             return ((SpongeGenerationPopulator) this.baseGenerator).getHandle(this.world).getNearestStructurePos(worldIn, structureName, position,
                     p_180513_4_);
         }
+
         return null;
     }
 
     @Override
     public boolean isInsideStructure(World worldIn, String structureName, BlockPos position) {
+        if (this.baseGenerator instanceof SpongeGenerationPopulator) {
```

Comment:
Fix this one too as per `GetNearestStructurePos`

Reply:
Updated the PR to match.  Also removed the below return though I don't have a plugin that is doing structure generation where I can test that aspect of this PR.